src/components/ui/Tiles/Tiles.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { forwardRef } from 'react'
+import type { HTMLAttributes } from 'react'
+
+import Tile from './Tile'
+
+export interface TilesProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress,
+   * testing-library, and jest).
+   */
+  testId?: string
+}
+
+const MIN_CHILDREN = 2
+const MAX_CHILDREN = 4
+const NUMBER_ITEMS_TO_EXPAND = 3
+
+const Tiles = forwardRef<HTMLDivElement, TilesProps>(function Tiles(
+  { testId = 'store-tiles', children, ...otherProps },
+  ref
+) {
+  const childrenCount = React.Children.count(children)
+  const spanFirstChild = childrenCount === NUMBER_ITEMS_TO_EXPAND
+  const isOutOfBounds =
+    childrenCount < MIN_CHILDREN || childrenCount > MAX_CHILDREN
+
+  if (isOutOfBounds) {
+    throw new Error(
+      `Tiles cannot receive less than ${MIN_CHILDREN} or more than ${MAX_CHILDREN} children.`
+    )
+  }
+
+  React.Children.forEach(children, (child) => {
+    if (!React.isValidElement(child) || child.type !== Tile) {
+      throw new Error('Only Tile components allowed as children.')
+    }
```

Comment:
Is it possible to define the type of the children as an array of Tile so there's no need to check this at runtime?

Reply:
I tried to do that and also have read some [threads](https://github.com/microsoft/TypeScript/issues/13618#issuecomment-307701505) about that. But it looks like we can't enforce to just have Tile's Array as children.

Reply:
At least move this check inside the `map` statement so we don't loop twice over the children

Reply:
Good suggestion, thanks. 
I'm gonna change the implementation (remove the `map`) so that we just need to loop once to check the `child`'s type.